src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -714,6 +730,15 @@ PeerImp::onMessage (std::shared_ptr <protocol::TMHello> const& m)
 }
 
 void
+PeerImp::onMessage (std::shared_ptr<protocol::TMManifests> const& m)
+{
+    // VFALCO What's the right job type?
+    getApp().getJobQueue().addJob (jtVALIDATION_ut,
+        "receiveManifests", std::bind(&OverlayImpl::onManifests,
+            &overlay_, std::placeholders::_1, m, shared_from_this()));
+}
+
```

Comment:
This is significantly inferior to the way we handle other signed objects like transactions, validations, and proposals. I'm concerned that we'll wind up with all our threads blocked in these jobs, waiting for the manifest lock which is held while we check the signature.

At an absolute minimum, dispatch these as their own job type with a concurrency limit of one. But I'd much rather see the same mechanism we use for other such objects used for those. For example, just by using the HashRouter, we can avoid dispatching every copy of a manifest we receive and we can avoid broadcasting the manifest to servers we receive it from.